### PR TITLE
kdeApplications.koi: init at 0.3.1

### PR DESCRIPTION
**Pull request by fnune**
## Description of changes

image

[Koi](https://github.com/baduhai/Koi) is a scheduling theme switcher for KDE that I have been using on Plasma 6 for the last few days. It has been working great for me, and I was originally looking for it within `nixpkgs`, but couldn't find it, so here it is.

This is the derivation I have been testing it with: https://github.com/fnune/home.nix/commit/7edb3f60a8ee767268ac2a9915f8720d745802f0

I have also been running it from my own development fork of `nixpkgs` using the same code I'm submitting here.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/kde/third-party/koi/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  kconfig,
+  kcoreaddons,
+  kwidgetsaddons,
+  wrapQtAppsHook,
+}:
+stdenv.mkDerivation rec {
+  pname = "koi";
+  version = "0.3.1";
+
+  meta = with lib; {
```
Comment (K900) [inline pkgs/kde/third-party/koi/default.nix:15]: The convention is to put meta last, generally

File: pkgs/kde/third-party/koi/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  };
+
+  # See https://github.com/baduhai/Koi/blob/master/development/Nix%20OS/dev.nix
+  sourceRoot = "source/src";
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [
+    kconfig
+    kcoreaddons
+    kwidgetsaddons
+    wrapQtAppsHook
```
Comment (K900) [inline pkgs/kde/third-party/koi/default.nix:51]: wrapQtAppsHook goes into nativeBuildInputs

**Comment by K900** [comment]
Overall looks OK modulo a few small nitpicks. Also, can you split the maintainer list update into a separate commit?

**Comment by fnune** [comment]
@K900 Different commit, or different PR?

**Comment by K900** [comment]
Different commit in the same PR is fine.

**Comment by K900** [comment]
You could just add yourself to maintainers first, and then add the package :)

**Comment by K900** [comment]
@ofborg build kdePackages.koi

**Comment by fnune** [comment]
> You could just add yourself to maintainers first, and then add the package :)

This seemed more natural, I don't know :D I can redo it if necessary.

**Comment by fnune** [comment]
I just tested the latest changes on my system, things still work fine.

**Comment by K900** [comment]
Doesn't really matter :)

**Pull request merged**